dask/dataframe/io/parquet.py

```diff
@@ -425,6 +426,19 @@ def to_parquet(path, df, compression=None, write_index=None, has_nulls=None,
 
     if len(fmd.row_groups) == 0:
         raise ValueError("All partitions were empty")
+    cats = {}
+    for col in df._meta:
+        if str(df[col].dtype) != 'category':
+            continue
+        for rg in fmd.row_groups:
+            for chunk in rg.columns:
+                if chunk.meta_data.path_in_schema[-1] == col:
+                    cats[col] = max(
+                        cats.get(col, 0),
+                        int(chunk.meta_data.key_value_metadata[0].value))
+    if cats:
+        fmd.key_value_metadata = [fastparquet.parquet_thrift.KeyValue(
+            key="fastparquet.cats", value=json.dumps(cats))]
```

Comment:
While I would like to see this fixed, I also think that it would be reasonable to wait a bit and see if we can't work with the Parquet folks on a standard solution.  If we can agree on a common format quickly then this saves us from having to support our own custom conventions alongside the standard ones into the future.